src/cmd/services/m3dbnode/config/config.go

```diff
@@ -387,6 +387,14 @@ type IndexConfiguration struct {
 	// as they are very CPU-intensive (regex and FST matching).
 	MaxQueryIDsConcurrency int `yaml:"maxQueryIDsConcurrency" validate:"min=0"`
 
+	// MaxResultsPerPermit is the maximum index results a query can process after obtaining a permit. If a query needs
+	// to process more results it yields the permit and must wait again for another permit to resume. The number of
+	// permits available to all queries is defined by MaxQueryIDsConcurrency.
+	// Capping the maximum results per permit ensures a few large queries don't hold all the concurrent permits and lock
+	// out many small queries from running. This should be set higher than the max results returned by the vast majority
+	// of queries, so most queries only need to obtain a single permit.
+	MaxResultsPerPermit int `yaml:"maxResultsPerPermit"`
```

Comment:
open to suggestions on this name. I didn't like adding `Permit` to the name, since that feels like an impl detail. However, I didn't really know how else to describe it.

Reply:
I think this is fine. We could do `MaxResultsPerAcquireWorker` which is also impl. specific but decoupled from the concept of permits? Either way though I don't think it's bad, in general more folks would understand what a "worker" is than a "permit" but up to what you think is best here.